recipes/setup_clustering.rb

```diff
@@ -0,0 +1,82 @@
+#
+# Cookbook Name:: splunk
+# Recipe:: setup_clustering
+#
+# Author: Roy Arsan <rarsan@splunk.com>
+# Copyright (c) 2014, Chef Software, Inc <legal@chef.io>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+unless node['splunk']['clustering']['enable']
+  Chef::Log.debug('The chef-splunk::setup_clustering recipe was added to the node,')
+  Chef::Log.debug('but the attribute to enable clustering was not set.')
+  return
+end
+
+# ensure that the splunk service resource is available without cloning
+# the resource (CHEF-3694). this is so the later notification works,
+# especially when using chefspec to run this cookbook's specs.
+begin
+  resources('service[splunk]')
+rescue Chef::Exceptions::ResourceNotFound
+  service 'splunk'
+end
+
+include_recipe 'chef-vault'
+
+passwords = chef_vault_item('vault', "splunk_#{node.chef_environment}")
+splunk_auth_info = passwords['auth']
+
+cluster_secret = passwords['secret']
+cluster_params = node['splunk']['clustering']
+cluster_mode = cluster_params['mode']
+
+Chef::Log.info("Current node clustering mode: #{cluster_mode}")
```

Comment:
It also can be moved up to the top, as it will print this message during the recipe evaluation (compile) phase.